devices/ePaper/Drivers/Gdew0154x/Gdew0154x.cs

```diff
@@ -0,0 +1,636 @@
+
+            SendCommand((byte)Command.PowerOn);
+
+            WaitReady();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the intenral frame buffer. Supports paging.
+        /// </summary>
+        /// <param name="width">The width of the frame buffer in pixels.</param>
+        /// <param name="height">The height of the frame buffer in pixels.</param>
+        /// <param name="enableFramePaging">If <see langword="true"/>, enables paging the frame.</param>
+        protected virtual void InitializeFrameBuffer(int width, int height, bool enableFramePaging)
+        {
+            var frameBufferHeight = enableFramePaging ? height / PagesPerFrame : height;
+            FrameBuffer1bpp = new FrameBuffer1BitPerPixel(frameBufferHeight, width);
+        }
+
+        /// <inheritdoc/>
+        public virtual bool NextFramePage()
+        {
+            if (PagedFrameDrawEnabled && CurrentFrameBufferPage < PagesPerFrame - 1)
+            {
+                Flush();
+
+                CurrentFrameBufferPage++;
+                SetFrameBufferPage(CurrentFrameBufferPage);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public virtual void PerformFullRefresh()
+        {
+            WaitMs(10);
+            SendCommand((byte)Command.DisplayRefresh);
+            WaitReady();
+        }
+
+        /// <inheritdoc/>
+        public virtual void PerformPartialRefresh()
+        {
+            // No partial refresh implemented
+            PerformFullRefresh();
+        }
+
+        /// <summary>
+        /// Puts the display to sleep using the specified <see cref="SleepMode"/>.
+        /// </summary>
+        /// <param name="sleepMode">The sleep mode to use when powering down the display.</param>
+        public virtual void PowerDown(SleepMode sleepMode = SleepMode.Normal)
+        {
+            SendCommand(0x50);
+            SendData(0x07);
+            SendCommand((byte)Command.PowerOff);
+            WaitReady();
+
+            if (sleepMode == SleepMode.DeepSleepMode)
+            {
+                SendCommand((byte)Command.DeepSleepMode);
+                SendData(0xa5);
+            }
+
+            PowerState = PowerState.PoweredOff;
+        }
+
+        /// <summary>
+        /// Performs the required steps to "power on" the display.
+        /// </summary>
+        public virtual void PowerOn()
+        {
+            HardwareReset();
+            WaitMs(1000);
+            SendCommand(0x50);
+            SendData(0xd7);
+            SendCommand((byte)Command.PowerOn);
+            WaitReady();
+
+            PowerState = PowerState.PoweredOn;
+        }
+
+        /// <inheritdoc/>
+        public virtual void SendCommand(params byte[] command)
+        {
+            // make sure we are setting data/command pin to low (command mode)
+            _dataCommandPin.Write(PinValue.Low);
+
+            foreach (var b in command)
```

Comment:
maybe for performance reasons, it could be faster to write all the bytes at once? As every time, the data is sent independently with a low and high on the chip select. So it's much slower than sending in a single write with a buffer.